shap/plots/_scatter.py

```diff
@@ -248,49 +244,9 @@ def scatter(
     categorical_interaction = False
 
     # create a matplotlib figure, if `ax` hasn't been specified.
-    if not ax:
+    if ax is None:
```

Comment:
we are doing the opposite here than compared to the previous version. I don't see how the previous case (including the recursion) is handled. Unfortunately I believe the plots are not really well covered by tests.